clang/lib/Driver/Driver.cpp

```diff
@@ -4482,55 +4620,77 @@ void Driver::BuildActions(Compilation &C, DerivedArgList &Args,
       UnbundlerInputs.push_back(LI);
     }
     const Arg *LastArg;
-    auto addUnbundlerInput = [&](types::ID T, const Arg *A) {
+    auto addUnbundlerInput = [&](types::ID T, const char *A) {
       const llvm::opt::OptTable &Opts = getOpts();
-      Arg *InputArg = MakeInputArg(Args, Opts, A->getValue());
+      Arg *InputArg = MakeInputArg(Args, Opts, C.getArgs().MakeArgString(A));
       LastArg = InputArg;
       Action *Current = C.MakeAction<InputAction>(*InputArg, T);
       UnbundlerInputs.push_back(Current);
     };
-    for (const auto *A : Args.filtered(options::OPT_foffload_static_lib_EQ))
-      addUnbundlerInput(types::TY_Archive, A);
-    for (const auto *A :
-        Args.filtered(options::OPT_foffload_whole_static_lib_EQ))
-      addUnbundlerInput(types::TY_WholeArchive, A);
+    bool IsWholeArchive = false;
+    for (std::string MA : LinkArgs) {
+      if (isStaticArchiveFile(MA)) {
+        addUnbundlerInput(IsWholeArchive ? types::TY_WholeArchive
+                                         : types::TY_Archive,
+                          MA.c_str());
+        continue;
+      }
+      if (optionMatches("-no-whole-archive", MA)) {
+        IsWholeArchive = false;
+        continue;
+      }
+      if (optionMatches("-whole-archive", MA)) {
+        IsWholeArchive = true;
+        continue;
+      }
+      if (isObjectFile(MA)) {
+        // Add any objects to the unbundler step.  These objects are passed
+        // directly to the linker, so the driver does not know about them.
+        // FIXME - Better process objects passed to the linker process them
```

Comment:
```suggestion
        // FIXME - Better process objects passed to the linker
```
What could become/is a use-case here?

Reply:
Improved comment.

Reply:
How important is this usecase, BTW?

Reply:
It only applies for folks who use -Wl,<objects> as opposed to using the command line directly to compile objects.  I don't know how common it is, but the workaround is pretty straight-forward.  There are definitely nuances we need to understand if we start processing these objects as we can't pass along unbundled host objects to the host link as it will cause duplication.